App/Lettuce/Sources/Lettuce/MessageCenter/Requests/messaging/messages/WriteNewMessageRequest.swift

```diff
@@ -0,0 +1,29 @@
+// system
+import Foundation
+import HTTPTypes
+
+// local
+import Onion
+
+/// https://developer.allegro.pl/documentation#operation/newMessagePOST
+struct WriteNewMessageRequest: Request {
```

Comment:
That should be a `UploadRequest` and then you will have to define the `body` property :)

You can check the `ChangeReadFlagOnThreadRequest` for inspiration. In this example, it looks like `NewMessage` would be the inner type called `Body`. 

That is something that I'm struggling with. When it's an inner type then it's tied more closely to the request type. When it's standalone, it can be reused in other "contexts". Not that one is better than the other one but there are tradeoffs :D  I'm fine with being a stand-alone type :)